### PR TITLE
Add ThirdReality vibration sensor

### PR DESCRIPTION
**Pull request by weihuan1111**
Change-Id: Ia5ba0d3c10c86c0f3277b483e797dd4f9c59d424

## Checklist

- [x] The changes are tested and work correctly
- [ ] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

File: zhaquirks/thirdreality/vibrate.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    cluster_id = MANUFACTURER_SPECIFIC_CLUSTER_ID
+    attributes = {
+        0x0001: ("x_axis", t.int16s, True),
+        0x0002: ("y_axis", t.int16s, True),
+        0x0003: ("z_axis", t.int16s, True),
+    }
+
+class Vibrate(CustomDevice):
+    """thirdreality vibrate device - alternate version."""
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/vibrate.py]: I couldn't find a previous version of this device, so I think this isn't an "alternate version"?
```suggestion
    """thirdreality vibrate device."""
```

File: zhaquirks/thirdreality/vibrate.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    signature = {
+        MODELS_INFO: [(THIRD_REALITY, "3RVS01031Z")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: 0x0104,
+                DEVICE_TYPE: 0x0402,
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/vibrate.py]: We can probably use the [constant for `IAS_ZONE`](https://github.com/zigpy/zigpy/blob/ec5b3cd26573e21773a2c6e37f9c3700701a8ec0/zigpy/profiles/zha.py#L62) then:
```suggestion
                DEVICE_TYPE: zha.DeviceType.IAS_ZONE,
```
(Note: `zha` profile probably needs to be imported then: `from zigpy.profiles import zha`)

File: zhaquirks/thirdreality/vibrate.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+class Vibrate(CustomDevice):
+    """thirdreality vibrate device - alternate version."""
+
+    signature = {
+        MODELS_INFO: [(THIRD_REALITY, "3RVS01031Z")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: 0x0104,
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/vibrate.py]: Use the [ZHA profile id constant](https://github.com/zigpy/zigpy/blob/ec5b3cd26573e21773a2c6e37f9c3700701a8ec0/zigpy/profiles/zha.py#L5) here:
```suggestion
                PROFILE_ID: zha.PROFILE_ID,
```
(Note: `zha` profile probably needs to be imported then: ``from zigpy.profiles import zha``)

File: zhaquirks/thirdreality/vibrate.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                OUTPUT_CLUSTERS: [
+                    Ota.cluster_id,
+                ],
+            }
+        },
+    }
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                DEVICE_TYPE: 0x0402,
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/vibrate.py]: `PROFILE_ID` is missing in the `replacement` (and also use constant for `DEVICE_TYPE`).
```suggestion
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.IAS_ZONE,
```

File: zhaquirks/thirdreality/vibrate.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+class ThirdRealityAccelCluster(CustomCluster):
+    """ThirdReality Acceleration Cluster."""
+
+    cluster_id = MANUFACTURER_SPECIFIC_CLUSTER_ID
+    attributes = {
+        0x0001: ("x_axis", t.int16s, True),
+        0x0002: ("y_axis", t.int16s, True),
+        0x0003: ("z_axis", t.int16s, True),
+    }
```
Comment (TheJulianJES) [inline zhaquirks/thirdreality/vibrate.py:30]: 1. So for now, these attributes could only be read using the "clusters UI" in ZHA, right?
2. Do you plan on exposing these custom attributes in the HA UI [by adding a new match/sensor in ZHA (at the bottom of the file)](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zha/sensor.py)? (And also binding the cluster by adding it to the ["manufacturer specific" cluster handler](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zha/core/cluster_handlers/manufacturerspecific.py)?
3. Does the `IasZone` cluster on this device do anything? So "on/off" for vibrations?

**Review by TheJulianJES — CHANGES_REQUESTED** [review]

File: zhaquirks/thirdreality/vibrate.py (hunk 5 above)
Comment (weihuan1111) [reply zhaquirks/thirdreality/vibrate.py:30]: @TheJulianJES  now, these attributes could only be read using the "clusters UI" in ZHA is ok, we will try exposing these custom attributes in the HA UI later.
About Iaszone cluster, when the device vibrate, will send accelcluster and iaszone to show the vibration

**Review by TheJulianJES — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**